frontroomsbot/cogs/tldr.py

```diff
@@ -1,79 +1,214 @@
 import discord
-from discord import app_commands, MessageType
+from discord import (
+    app_commands,
+    MessageType,
+    TextChannel,
+    Message,
+    Interaction,
+    AppCommandType,
+)
 from discord.ext import commands
 from bot import BackroomsBot
 import google.generativeai as genai
 import re
+import json
+from collections import defaultdict
 
 from consts import GEMINI_TOKEN
 
+
 USER_RE = re.compile(r"<@\d+>")
 
 
+class TldrError(Exception):
+    def __init__(self, error_msg: str):
+        self.error_msg = error_msg
+        super().__init__(self.error_msg)
+
+
+class TokensLimitExceededError(TldrError):
+    pass
+
+
+class MessageIdInvalidError(TldrError):
+    pass
+
+
+class StartMsgOlderThanEndMsgError(TldrError):
+    def __init__(
+        self,
+        error_msg: str = "Starting message must not be newer than the ending message.",
+    ):
+        super().__init__(error_msg)
+
+
 class TldrCog(commands.Cog):
```

Comment:
this could be done via ConfigCog, all of these consts could be configured via that